### ISSUE TITLE
AZURE_SEARCH_INDEX_TOP_K / AZURE_SEARCH_TOP_K have no effect

### ISSUE THREAD

**Issue by ClaudiaNightScout** [microsoft/sample-app-aoai-chatGPT#312]
Chat still only returns 5 references

**Comment by eivholt** [comment]
There are some errors in the names of the configs when deploying as a web app, for instance "AZURE_SEARCH_TOP_K" is named "SEARCH_TOP_K" and the code in app.py will not find your value and default to 5. The same applies for at least one other key, can't remember which one.

**Comment by ClaudiaNightScout** [comment]
I have put SEARCH_TOP_K and AZURE_SEARCH_TOP_K  and AZURE_SEARCH_INDEX_TOP_K up to 20 but still no change.

**Comment by ClaudiaNightScout** [comment]
> I have put SEARCH_TOP_K and AZURE_SEARCH_TOP_K and AZURE_SEARCH_INDEX_TOP_K up to 20 but still no change.

Ok now im getting this error, redeployed and added values.

Error: Request body validation error. Reason: Validator type=integer failed on object at path '$.dataSources[0].parameters.topNDocuments'

**Comment by eivholt** [comment]
It might expect the number as a string, under Configuration in the web app, check advanced settings and make sure the value is encased in quotes:
{
    "name": "AZURE_SEARCH_TOP_K",
    "value": "10",
    "slotSetting": false
  }

**Comment by ClaudiaNightScout** [comment]
> It might expect the number as a string, under Configuration in the web app, check advanced settings and make sure the value is encased in quotes: { "name": "AZURE_SEARCH_TOP_K", "value": "10", "slotSetting": false }

In quotes response

Error: Request body validation error. Reason: Validator type=integer failed on object at path '$.dataSources[0].parameters.topNDocuments', Validator format=int32 failed on object at path '$.dataSources[0].parameters.topNDocuments

**Comment by kitfit-dave** [comment]
I think this might be something that has changed somewhere on the Azure side, we are getting "Request body validation error" on a project that was working just fine up into a couple of days ago, and has not been modified in any way.

**Comment by BruceEx** [comment]
Agree with @kitfit-dave . Seems like a bug in Azure web app Environment variables. The issue immediately happens while nothing changed in the code after working some days. After remove topk environment variable in Azure, it can work now.

**Comment by ClaudiaNightScout** [comment]
Yes, remove AZURE_SEARCH_TOP_K from Azure config and its all working apart from there is no mechanism to change top_k

**Comment by eivholt** [comment]
The environment variable keys that need to be changed after deployment from Playground are SEARCH_TOP_K, SEARCH_STRICTNESS and SEARCH_ENABLE_IN_DOMAIN. You can verify this by looking at the # ACS Integration Settings in app.py. Here you will see that if you have a wrong or missing environment variable, for instance AZURE_SEARCH_TOP_K will have a default value, (5 if i remember correctly). I just verified that this works by redeploying and getting a limited number of hits, then renaming and setting the value to 20 and getting an expected larger number of hits.

**Comment by ClaudiaNightScout** [comment]
> The environment variable keys that need to be changed after deployment from Playground are SEARCH_TOP_K, SEARCH_STRICTNESS and SEARCH_ENABLE_IN_DOMAIN. You can verify this by looking at the # ACS Integration Settings in app.py. Here you will see that if you have a wrong or missing environment variable, for instance AZURE_SEARCH_TOP_K will have a default value, (5 if i remember correctly). I just verified that this works by redeploying and getting a limited number of hits, then renaming and setting the value to 20 and getting an expected larger number of hits.

HI, just to confirm in your setup you only have
SEARCH_STRICTNESS                          1
SEARCH_ENABLE_IN_DOMAIN            true
SEARCH_TOP_K                                    20


AZURE_SEARCH_TOP_K  is deleted or set to 20?

**Comment by eivholt** [comment]
If you want to use anything besides default values you need to add "AZURE_" in front of each of the environment variables SEARCH_STRICTNESS 
SEARCH_ENABLE_IN_DOMAIN,
SEARCH_TOP_K 20, SEARCH_TOP_K. Currently, when deploying a web-app from OAI Playground these are incorrectly named.

**Comment by cristhianvs** [comment]
I don't know if you tried in this way:
"topNDocuments": int(AZURE_SEARCH_TOP_K),
I mean forcing to Integer type
